Refactor cancel_analysis

henrikstranneheim opened the pull request:
## Description

Refactor cancel analysis

### Added

- Tests

### Changed

- Refactor and migrate cancel_analysis to CRUD

### How to prepare for test
- [ ] ssh to hasta (depending on type of change)
- [ ] activate stage: `us`
- [ ] request trailblazer-stage on hasta: `paxa`
- [ ] install on stage:
    ```Shell
    bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-tool-stage.sh -e S_trailblazer -t trailblazer -b refactor_cancel_analysis -a
    ```
- [ ] ssh to clinical-db (depending on type of change)
- [ ] install on stage:
`bash /home/proj/production/servers/resources/clinical-db.scilifelab.se/update-trailblazer-ui-stage.sh [THIS-BRANCH-NAME]`

### How to test
- [ ] login to ...
- [ ] do ...

### Expected test outcome
- [ ] check that ...
- [ ] Take a screenshot and attach or copy/paste the output.

## Review
- [ ] tests executed by
- [ ] "Merge and deploy" approved by
Thanks for filling in who performed the code review and the test!

### This [version](https://semver.org/) is a
- [ ] **MAJOR** - when you make incompatible API changes
- [ ] **MINOR** - when you add functionality in a backwards compatible manner
- [x] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

islean reviewed (approved):
Looks good! I think we could consider not raising an error if there is an analysis which is not ongoing since that seems to be what we are after but feel to keep it as is as well!

henrikstranneheim commented:
### CLI and API ✅ 
```
Action running set for case justhusky
mip-dna run started!
θ61° [henrik.stranneheim@hasta:~] [S_base] 1m42s $ trailblazer scan
Updated status justhusky - 278331: running
Status in Tower: hardygrouse - 278330 - 4zY4HUy9hew76p
Updated status hardygrouse - 278330: running
Error updating analysis for: case - civilsole : FileNotFoundError
Error updating analysis for: case - savingstag : FileNotFoundError
Error updating analysis for: case - HW2V2DRX2 : FileNotFoundError
Error updating analysis for: case - divinetomcat : PermissionError
Error updating analysis for: case - handyshrimp : FileNotFoundError
Error updating analysis for: case - setcolt : PermissionError
Error updating analysis for: case - proudfish : PermissionError
Error updating analysis for: case - funnyaphid : FileNotFoundError
Error updating analysis for: case - onestag : PermissionError
Error updating analysis for: case - newgopher : PermissionError
Error updating analysis for: case - helpedlab : MissingSqueueOutput
Error updating analysis for: case - goodbengal : MissingSqueueOutput
Error updating analysis for: case - grandaphid : PermissionError
```

<img width="1424" alt="Screenshot 2023-09-12 at 08 48 36" src="https://github.com/Clinical-Genomics/trailblazer/assets/584738/774c52df-25fc-4354-9858-bbd9a8040425">

```
θ62° [henrik.stranneheim@hasta:~] [S_base] 4s 130 $ trailblazer cancel 278331
Cancelling job 5130933 - bwa_mem2
Cancelling job 5130934 - bwa_mem2
Cancelling job 5130932 - bwa_mem2
Cancelling job 5130935 - samtools_merge
Cancelling job 5130936 - samtools_merge
Cancelling job 5130937 - samtools_merge
Cancelling job 5130938 - markduplicates
Cancelling job 5130939 - markduplicates
Cancelling job 5130940 - markduplicates
Cancelling job 5130966 - cnvnator_ar
Cancelling job 5130967 - cnvnator_ar
Cancelling job 5130968 - cnvnator_ar
Cancelling job 5130970 - manta
Cancelling job 5130974 - sv_varianteffectpredictor
Cancelling job 5130975 - sv_vcfparser
Cancelling job 5130976 - sv_rankvariant
Cancelling job 5130989 - deepvariant
Cancelling job 5130990 - deepvariant
Cancelling job 5130991 - deepvariant
Cancelling job 5130992 - glnexus_merge
Cancelling job 5131008 - prepareforvariantannotationblock
Cancelling job 5131009 - rhocall_ar
Cancelling job 5131010 - bcftools_norm
Cancelling job 5131011 - variant_annotation
Cancelling job 5131021 - frequency_filter
Cancelling job 5131022 - cadd_ar
Cancelling job 5131023 - varianteffectpredictor
Cancelling job 5131024 - vcfparser_ar
Cancelling job 5131025 - rankvariant
Cancelling job 5130941 - chanjo_sexcheck
Cancelling job 5130942 - chanjo_sexcheck
Cancelling job 5130943 - chanjo_sexcheck
Cancelling job 5130944 - samtools_subsample_mt
Cancelling job 5130945 - samtools_subsample_mt
Cancelling job 5130946 - samtools_subsample_mt
Cancelling job 5130947 - tiddit_coverage
Cancelling job 5130948 - tiddit_coverage
Cancelling job 5130949 - tiddit_coverage
Cancelling job 5130950 - chromograph_cov
Cancelling job 5130951 - chromograph_cov
Cancelling job 5130952 - chromograph_cov
Cancelling job 5130953 - mitodel
Cancelling job 5130954 - mitodel
Cancelling job 5130955 - mitodel
Cancelling job 5130956 - sambamba_depth
Cancelling job 5130957 - sambamba_depth
Cancelling job 5130958 - sambamba_depth
Cancelling job 5130959 - smncopynumbercaller
Cancelling job 5130960 - picardtools_collectmultiplemetrics
Cancelling job 5130961 - picardtools_collectmultiplemetrics
Cancelling job 5130962 - picardtools_collectmultiplemetrics
Cancelling job 5130963 - picardtools_collecthsmetrics
Cancelling job 5130964 - picardtools_collecthsmetrics
Cancelling job 5130965 - picardtools_collecthsmetrics
Cancelling job 5130969 - expansionhunter
Cancelling job 5130971 - tiddit
Cancelling job 5130972 - sv_combinevariantcallsets
Cancelling job 5130973 - sv_annotate
Cancelling job 5130977 - sv_reformat
Cancelling job 5130978 - vcf2cytosure_ar
Cancelling job 5130979 - me_merge_bam
Cancelling job 5130980 - me_merge_bam
Cancelling job 5130981 - me_merge_bam
Cancelling job 5130982 - retroseq
Cancelling job 5130983 - retroseq
Cancelling job 5130984 - retroseq
Cancelling job 5130985 - me_merge_vcfs
Cancelling job 5130986 - me_annotate
Cancelling job 5130987 - me_varianteffectpredictor
Cancelling job 5130988 - me_filter
Cancelling job 5130993 - gatk_combinevariantcallsets
Cancelling job 5130994 - peddy_ar
Cancelling job 5130995 - plink
Cancelling job 5130996 - gatk_variantevalall
Cancelling job 5130997 - gatk_variantevalall
Cancelling job 5130998 - gatk_variantevalall
Cancelling job 5130999 - gatk_collectreadcounts
Cancelling job 5131000 - gatk_collectreadcounts
Cancelling job 5131001 - gatk_collectreadcounts
Cancelling job 5131002 - gatk_denoisereadcounts
Cancelling job 5131003 - gatk_denoisereadcounts
Cancelling job 5131004 - gatk_denoisereadcounts
Cancelling job 5131005 - gens_generatedata
Cancelling job 5131006 - gens_generatedata
Cancelling job 5131007 - gens_generatedata
Cancelling job 5131012 - rhocall_viz
Cancelling job 5131013 - rhocall_viz
Cancelling job 5131014 - rhocall_viz
Cancelling job 5131015 - chromograph_rhoviz
Cancelling job 5131016 - chromograph_rhoviz
Cancelling job 5131017 - chromograph_rhoviz
Cancelling job 5131018 - upd_ar
Cancelling job 5131019 - chromograph_upd
Cancelling job 5131020 - mt_annotation
Cancelling job 5131026 - endvariantannotationblock
Cancelling job 5131027 - gatk_variantevalexome
Cancelling job 5131028 - gatk_variantevalexome
Cancelling job 5131029 - gatk_variantevalexome
Cancelling job 5131030 - qccollect_ar
Cancelling job 5131031 - version_collect_ar
Cancelling job 5131032 - multiqc_ar
Cancelling job 5131033 - analysisrunstatus
Cancelling job 5131034 - sacct
Cancelling job 5130930 - fastqc_ar
Cancelling job 5130931 - fastqc_ar
Cancelling job 5130929 - fastqc_ar
Case justhusky - Analysis 278331: all ongoing jobs cancelled successfully!
Updated status justhusky - 278331: running
```

<img width="1424" alt="Screenshot 2023-09-12 at 08 52 46" src="https://github.com/Clinical-Genomics/trailblazer/assets/584738/28ee7257-3592-4381-98d4-fabf85f989a4">

henrikstranneheim commented:
Deployed to Hasta:

```
Log deploy... done.
trailblazer, version 18.0.4
```
